scripts/swupdate/merge.sh

```diff
@@ -1,112 +1,3 @@
-          then
-            IFS=$'\n'
-            if [ "$fileType" = "properties" ]
-            then
-              sed -i "s+$key *= *$configValue+$key=$newSampleValue+" $configFile
-            else
-              sed -i "s+$key *: *$configValue+$key: $newSampleValue+" $configFile
-            fi
-          fi
-        fi
-      fi
-    fi
-  done < $newSampleFile
-echo "Merge complete for ${configFile}"
-}
-
-#Below method will delete temporarily created old files
-function performCleanUp {
-echo "Removing old sample file backup.."
-rm -f /tmp/s3config.yaml.sample.old
-rm -f /tmp/config.yaml.sample.old
-rm -f /tmp/authserver.properties.sample.old
-rm -f /tmp/keystore.properties.sample.old
-}
-
-#Script logic starts here
-if [ -f /opt/seagate/cortx/s3/conf/s3config.yaml -a -f /tmp/s3config.yaml.sample.old ]; then
-  configFile=/opt/seagate/cortx/s3/conf/s3config.yaml
-  oldSampleFile=/tmp/s3config.yaml.sample.old
-  newSampleFile=/opt/seagate/cortx/s3/conf/s3config.yaml.sample
-  unsafeAttributesFile=/opt/seagate/cortx/s3/conf/s3config_unsafe_attributes.yaml
-  fileType=yaml
-  performMerge
-else
-  cp /opt/seagate/cortx/s3/conf/s3config.yaml.sample /opt/seagate/cortx/s3/conf/s3config.yaml
-fi
-if [ -f /opt/seagate/cortx/s3/s3backgrounddelete/config.yaml -a -f /tmp/config.yaml.sample.old ]; then
-  configFile=/opt/seagate/cortx/s3/s3backgrounddelete/config.yaml
-  oldSampleFile=/tmp/config.yaml.sample.old
-  newSampleFile=/opt/seagate/cortx/s3/s3backgrounddelete/config.yaml.sample
-  unsafeAttributesFile=/opt/seagate/cortx/s3/s3backgrounddelete/s3backgrounddelete_unsafe_attributes.yaml
-  fileType=yaml
-  performMerge
-else
-  cp /opt/seagate/cortx/s3/s3backgrounddelete/config.yaml.sample /opt/seagate/cortx/s3/s3backgrounddelete/config.yaml
-fi
-if [ -f /opt/seagate/cortx/auth/resources/authserver.properties -a -f /tmp/authserver.properties.sample.old ]; then
-  configFile=/opt/seagate/cortx/auth/resources/authserver.properties
-  oldSampleFile=/tmp/authserver.properties.sample.old
-  newSampleFile=/opt/seagate/cortx/auth/resources/authserver.properties.sample
-  unsafeAttributesFile=/opt/seagate/cortx/auth/resources/authserver_unsafe_attributes.yaml
-  fileType=properties
-  performMerge
-else
-  cp /opt/seagate/cortx/auth/resources/authserver.properties.sample /opt/seagate/cortx/auth/resources/authserver.properties
-fi
-if [ -f /opt/seagate/cortx/auth/resources/keystore.properties -a -f /tmp/keystore.properties.sample.old ]; then
-  configFile=/opt/seagate/cortx/auth/resources/keystore.properties
-  oldSampleFile=/tmp/keystore.properties.sample.old
-  newSampleFile=/opt/seagate/cortx/auth/resources/keystore.properties.sample
-  unsafeAttributesFile=/opt/seagate/cortx/auth/resources/keystore_unsafe_attributes.yaml
-  fileType=properties
-  performMerge
-else
-  cp /opt/seagate/cortx/auth/resources/keystore.properties.sample /opt/seagate/cortx/auth/resources/keystore.properties
-fi
-
-performCleanUp
+python3.6 merge.py
```

Comment:
Do we need this script now? In the script that calls merge.sh can we call merge.py there?

Reply:
Had a discussion with nilesh last friday, the task was to only port the logic and not hook it up in rpm spec. When the provisioner interface is available this script will be merged into that interface. But since then a lot of things have changed like the unsafe attributes file so merge.sh would no longer be able to do the job of upgrading. I dont know if spec file can invoke a python script and there is no way to test it locally since upgrade works on h/w. Hence keeping merge.sh to invoke merge.py

Reply:
Found a way to remove merge.sh(and invoking py file from rpm spec) while checking out the miniprovisioning job failure.